### PR TITLE
Bug fix/skip and hard limit in modification

### PR DESCRIPTION
**Pull request by mchacki**
Fixed an issue with skip and hardLimit modification within subqueries.

Added Regression test.

Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10177/

### CODE REVIEW

File: tests/js/server/aql/aql-modify.js [hunk 1]
```diff
@@ -1290,12 +1330,14 @@ function aqlUpsertOptionsSuite() {
 /// @brief executes the test suites
 ////////////////////////////////////////////////////////////////////////////////
 
+/*
```
Comment (markuspf) [inline tests/js/server/aql/aql-modify.js:1333]: Is this intentional?

File: tests/js/server/aql/aql-modify.js [hunk 2]
```diff
@@ -1160,6 +1161,45 @@ function aqlUpsertOptionsSuite() {
...
+        LIMIT 14,13
+        RETURN {fv0, sq1}
+      `;
+      const res = db._query(q);
+      const { writesExecuted, writesIgnored } = res.getExtra().stats;
+      assertEqual(0, writesIgnored);
+      // We update every document once per subquery execution
+      assertEqual(3 * countBefore, writesExecuted);
+      assertEqual(0, res.toArray().length);
+      assertEqual(0, col.count()- countBefore, `Only updates no inserts`);
```
Comment (markuspf) [inline tests/js/server/aql/aql-modify.js:1200]:
```suggestion
      assertEqual(countBefore, col.count(), `Only updates no inserts`);
```

File: tests/js/server/aql/aql-modify.js (hunk 1 above)
Comment (mchacki) [reply tests/js/server/aql/aql-modify.js:1333]: oh noes. it is not.
Thanks for spotting!

**Comment by mchacki** [comment]
Rerun Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10181/

Now including the test-suite again.

**Comment by mchacki** [comment]
Jenkins Blue

**Pull request merged**